### ISSUE TITLE
Dependency insight: "Force" selection reason does not show up when there is a substitution in place

### ISSUE THREAD

**Issue by OdysseusLives** [nebula-plugins/gradle-nebula-integration#4]
## Scenario
Given a build file containing a dependency with a substitution and a force: 
```
dependencies {
    compile 'org.mockito:mockito-all:1.8.0'
}

configurations.all {
    resolutionStrategy {
        force 'org.mockito:mockito-all:1.10.17'
    }
}

def substitutionMessage = '✭ substitution org.mockito:mockito-all -> org.mockito:mockito-core:1.10.19'
configurations.all {
    resolutionStrategy.dependencySubstitution {
        substitute module('org.mockito:mockito-all') because (substitutionMessage) with module('org.mockito:mockito-core:1.10.19')
    }
}
```

when I run `./gradlew dependencyInsight --dependency mockito` 
with Gradle 4.10-20180801000007+0000

then I should see the final version as `1.10.19` (which I do)
and I should see a selection reason as `Forced` (which I do *not*)
**Note**: See full output below

## Example Project
Please see the whole example project at [dependency-insight/docs/basic/substitute-static-force/input](https://github.com/nebula-plugins/gradle-nebula-integration/tree/master/dependency-insight/docs/substitute/substitute-static-force/input)

----------

## Current output for reference
```
> Task :dependencyInsight
org.mockito:mockito-core:1.10.19
   variant "compile" [
      org.gradle.status = release (not requested)
      org.gradle.usage  = java-api
   ]
   Selection reasons:
      - Was requested
      - Selected by rule : ✭ substitution org.mockito:mockito-all -> org.mockito:mockito-core:1.10.19

org.mockito:mockito-all:1.8.0 -> org.mockito:mockito-core:1.10.19
\--- compileClasspath
```

**Comment by bigdaz** [comment]
This was fixed as part of gradle/gradle#6826, and can be tested with a recently nightly build.
For reference, here is the output using Gradle built from `master`:
```
> Task :dependencyInsight
org.mockito:mockito-core:1.10.19
   variant "compile" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-api
      org.gradle.component.category = library (not requested)
   ]
   Selection reasons:
      - Forced
      - Selected by rule : ✭ substitution org.mockito:mockito-all -> org.mockito:mockito-core:1.10.19

org.mockito:mockito-all:1.8.0 -> org.mockito:mockito-core:1.10.19
\--- compileClasspath
```

**Comment by OdysseusLives** [comment]
Looks good, thank you!
